src/common/Formatter.cc

```diff
@@ -326,6 +328,8 @@ void XMLFormatter::flush(std::ostream& os)
    * we should NOT output a newline. This primarily triggers on HTTP redirects */
   if (m_pretty && !m_ss_str.empty())
     os << "\n";
+  if (m_line_break_enabled)
```

Comment:
Nit: ```else if``` to avoid possibility of a double line break if in the future someone enables it and doesn't guard against pretty printing also being enabled